falcon_multiqc/commands/query.py

```diff
@@ -40,156 +40,123 @@
+@click.option('--select', multiple=True, default=["sample"], type=click.Choice(["sample", "batch", "cohort", "tool-metric"], case_sensitive=False), required=False, help="What to select on (sample_name, batch, cohort, tool), default is sample_name.")
 @click.option('--tool_metric', multiple=True, type=(str, str, str, str), required=False, help="Filter by tool, metric, operator and number, e.g. 'verifybamid AVG_DP < 30'.")
-@click.option('--select', multiple=True, default=["sample_name"], type=click.Choice(["sample_name", "batch", "cohort", "tool"], case_sensitive=False), required=False, help="What to select on (sample_name, batch, cohort, tool), default is sample_name.")
 @click.option('--batch', multiple=True, required=False, help="Filter by batch name: enter which batches e.g. AAA, BAA, etc.")
 @click.option('--cohort', multiple=True, required=False, help="Filter by cohort id: enter which cohorts e.g. MGRB, cohort2, etc.")
 @click.option('--multiqc', is_flag=True, required=False, help="Create a multiqc report (user must select only for sample_name if so).")
 @click.option('--csv', is_flag=True, required=False, help="Create a csv report.")
 @click.option("-o", "--output", type=click.Path(), required=True, help="Output directory where query result will be saved.")
 def cli(select, tool_metric, batch, cohort, multiqc, csv, output):
     """Query the falcon qc database by specifying what you would like to select on by using the --select option, and
-    to filter on either --tool_metric, --batch, or --cohort."""
+    what to filter on (--tool_metric, --batch, or --cohort)."""
 
-    if multiqc and "sample_name" not in select:
+    if multiqc and "sample" not in select:
         click.echo(
-            "When multiqc report is selected, please ensure to select for sample_name.")
+            "When multiqc report is selected, please ensure to select for sample.")
         sys.exit(1)
 
-    click.echo(f"Returning {select} by filtering for:")
-    [click.echo(f'{tm}') for tm in tool_metric]
-    [click.echo(f'{b}') for b in batch]
-    [click.echo(f'{c}') for c in cohort]
-
-    ### ================================= FILTER SECTION ==========================================####
-
-    # Having a global set which contains all these means you can perform any potential SELECT action (see the SELECT SECTION)
-    sample_query_set = set()
     # Keep track of the query header with query.column_descriptions. 
     # TODO: This should be updated if the query is altered.
     query_header = []
+    # Sqlaclehmy query that will be constructed based on this command's options.
+    falcon_query = None
+
+    ### ================================= SELECT  ==========================================####
+    with session_scope() as session:
+        falcon_query = query_select(session, select, tool_metric, multiqc)
 
+    ### ================================= FILTER  ==========================================####
+    ## 1. Tool - Metric
     if tool_metric:
```

Comment:
In order to make the changes above work, this needs to be simplified to:

    if tool_metric:
        with session_scope() as session:
            falcon_query = query_metric(session, falcon_query, tool_metric)